tests/test_curobo.py

```diff
@@ -26,93 +37,427 @@ def test_curobo():
-    robot.reset()
-    robot.keep_still()
-    og.sim.step()
-
-    successes, traj_paths = cmg.compute_trajectories(
-        target_pos=th.stack(eef_positions, dim=0),
-        target_quat=th.stack(eef_quats, dim=0),
-        is_local=True,
-        max_attempts=1,
-        enable_finetune_trajopt=True,
-        return_full_result=False,
-        success_ratio=1.0,
-        attached_obj=None,
-    )
-
-    # Execute the trajectory and make sure there's rarely any collisions
-    assert th.sum(successes) > 0.95, f"Failed to find > 95% collision-free trajectories: {successes}"
-    print(f"Total successes: {th.sum(successes)} / {len(successes)}")
-    for success, traj_path in zip(successes, traj_paths):
-        if not success:
-            continue
-        q_traj = cmg.path_to_joint_trajectory(traj_path)
-        for q in q_traj:
+
+        for _ in range(5):
+            og.sim.step()
+
+        env.scene.update_initial_state()
+        env.scene.reset()
+
+        # Create CuRobo instance
+        batch_size = 10
+        n_samples = 30
+
+        cmg = CuRoboMotionGenerator(
+            robot=robot,
+            batch_size=batch_size,
+            debug=False,
+            use_cuda_graph=True,
+            use_default_embodiment_only=True,
+        )
+
+        # Sample values for robot
+        th.manual_seed(1)
+        lo, hi = robot.joint_lower_limits.clone().view(1, -1), robot.joint_upper_limits.clone().view(1, -1)
+
+        if isinstance(robot, HolonomicBaseRobot):
+            lo[0, :2] = -0.1
+            lo[0, 2:5] = 0.0
+            lo[0, 5] = -math.pi
+            hi[0, :2] = 0.1
+            hi[0, 2:5] = 0.0
+            hi[0, 5] = math.pi
+
+        random_qs = lo + th.rand((n_samples, robot.n_dof)) * (hi - lo)
+
+        # Test collision with the environment (not including self-collisions)
+        collision_results = cmg.check_collisions(q=random_qs)
+
+        target_pos, target_quat = defaultdict(list), defaultdict(list)
+
+        floor_plane_prim_paths = {child.GetPath().pathString for child in og.sim.floor_plane._prim.GetChildren()}
+
+        # View results
+        false_positive = 0
+        false_negative = 0
+
+        target_pos_in_world_frame = defaultdict(list)
+        for i, (q, curobo_has_contact) in enumerate(zip(random_qs, collision_results)):
+            # Set robot to desired qpos
             robot.set_joint_positions(q)
             robot.keep_still()
-            og.sim.step()
-            assert len(robot.contact_list()) == 0
+            og.sim.step_physics()
+
+            # To debug
+            # cmg.save_visualization(robot.get_joint_positions(), "/scr/chengshu/Downloads/test.obj")
+
+            # Sanity check in the GUI that the robot pose makes sense
+            for _ in range(10):
+                og.sim.render()
+
+            # Validate that expected collision result is correct
+            touching_object = robot.states[Touching].get_value(obj)
+            touching_floor = False
```

Comment:
is this needed? this gets immediately overwritten later

Reply:
Deleted.